### PR TITLE
Upgrade to BMI 2.0

### PR DESCRIPTION
**Pull request by sverhoeven**
TODO
* [x] import bmipy instead of basic_modelling_interface
* [x] centralize starting container
* [x] Fix tests
* ~[x] Update docs~ No docs found using BMI<v2
* [x] Install grpc4bmi from pypy >=0.3 instead of GH branch, 
  * requires PR https://github.com/eWaterCycle/grpc4bmi/pull/124 merged and new version released.
* [x] Merge #334 and #324 into this branch.

### CODE REVIEW

File: src/ewatercycle/container.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def start_container(
+    work_dir: Union[str, Path],
+    image_engine: ImageForContainerEngines,
+    input_dirs: Optional[Iterable[str]] = None,
+    image_port=55555,
+    timeout=None,
+    delay=0,
+) -> Bmi:
```
Comment (Peter9192) [inline src/ewatercycle/container.py:34]: Having this isolated from the models is great!
Would it be possible to separate the image selection from the actual starting of container? I'd like to be able to start a container by simply passing in an image (assuming that I know which container engine I'm using)

**Review by Peter9192 — COMMENTED** [review]
Very nice to have the containers centralized. Haven't tested yet, but the code changes look good.

**Comment by Peter9192** [comment]
Also could you add a changelog entry?

**Comment by Peter9192** [comment]
Note that I've made #340 which solves merge conflicts between this branch and #324

File: src/ewatercycle/container.py (hunk 1 above)
Comment (sverhoeven) [reply src/ewatercycle/container.py:34]: I added start_apptainer_container() and start_docker_container()

**Comment by Peter9192** [comment]
Merged as part of #340

**Pull request closed without merging**